_posts/2025-09-30-Introducing-Derived-Source.markdown

```diff
@@ -0,0 +1,161 @@
+
+When configuring an index with derived source, OpenSearch validates that all fields are of supported types. When documents are accessed through operations such as search, update, or recovery, the derived source reconstructs each field’s value from either `doc_values` or stored fields as defined in the index mapping. Because this requires reading data from each field’s disk location rather than a single `_source` fetch, you may notice some latency degradation when retrieving large numbers of documents.
+
+
+## How to configure derived source?
+
+You can configure derived source at the index level. Because it changes the default behavior of storing the original `_source`, this setting can’t be updated after the index is created. This restriction prevents mixed behavior between the original stored source and the dynamically generated source (which can appear similar to the original source in output).
+
+To configure derived source, set `derived_source.enabled` to `true` in the index settings:
+
+```json
+PUT sample-index1
+{
+  "settings": {
+    "index": {
+      "derived_source": {
+        "enabled": true
+      }
+    }
+  },
+  "mappings": {
+    <index fields>
+  }
+}
+```
+
+For more information, see [Derived source](https://docs.opensearch.org/latest/field-types/metadata-fields/source/#derived-source).
+
+## Performance benchmarks
+
+Based on experiment runs, derived source can provide significant storage reduction for certain workloads, shown in the following table.
+
+Workload | Storage reduction
+:--- | :---
+nyc_taxis | 41% |
+http logs | 43% |
+elb logs | 58% |
+
+Search latency showed a regression ranging from 10% (for 1k documents in a terms aggregation) to 100% (for 10k documents in a match-all query). In some queries, however, latency improved, since reading from `doc_values` often avoids the decompression required when accessing the stored `_source` field. 
+
+Across these benchmarks, we observed significant indexing throughput improvements of up to 18%, along with a reduction in merge time ranging from 20% to 48%. This is due to lower CPU overhead when generating optimized segments, which also helps reduce merge overhead.
+
+With a reduced index size, additional benefits become apparent: smaller shards enable faster recovery during node restarts or shard relocations, and smaller segments require less disk I/O and fewer page cache swaps, resulting in more efficient queries.
+
+While file-based recovery remains fast, operation-based recovery can be slower due to the need to regenerate the `_source`. There are two types of operation-based recovery: 
+
+1. Lucene-based, which is impacted by document replication using derived source
+2. Translog-based, in which reading the original `_source` instead of regenerating it can still take up to twice as long because of how derived source handles documents in the translog.
```

Comment:
```suggestion
2. Translog based, in which reading the original `_source` instead of regenerating it can still take up to twice as long because of how derived source handles documents in the translog.
```